core/probe.cpp

```diff
@@ -1015,3 +1021,41 @@ void Probe::executeSignalCallback(const Func &func)
                 instance()->m_signalSpyCallbacks.constEnd(),
                 func);
 }
+
+SourceLocation Probe::objectCreationSourceLocation(QObject* object)
+{
+  if (!s_listener()->constructionBacktracesForObjects.contains(object)) {
+    IF_DEBUG(std::cout << "No backtrace for object available for object" << object << "." << std::endl;)
+    return SourceLocation();
+  }
+
+  backward::StackTrace &st = s_listener()->constructionBacktracesForObjects[object];
+  int distanceToQObject = 0;
+  m_traceResolver.load_stacktrace(st);
+
+  const QMetaObject *metaObject = object->metaObject();
+  while (metaObject != &QObject::staticMetaObject) {
+    distanceToQObject++;
+    metaObject = metaObject->superClass();
```

Comment:
This needs a nullptr check somewhere, as there are (rare) QObject instances whose super class chain does not terminate in a QObject (value type adaptors for QML for example).